src/Components/CreateImageWizard/formComponents/ActivationKeyInformation.js

```diff
@@ -0,0 +1,139 @@
+import { Button, Popover } from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+import {
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import api from '../../../api';
+
+const ActivationKeyInformation = () => {
+  const { getState } = useFormApi();
+
+  const activationKey = getState()?.values?.['subscription-activation-key'];
+  const [role, setRole] = useState(undefined);
+  const [serviceLevel, setServiceLevel] = useState(undefined);
+  const [usage, setUsage] = useState(undefined);
+  const [additionalRepositories, setRepositories] = useState(undefined);
+
+  useEffect(() => {
+    const fetchKeyInformation = async () => {
+      const data = await api.getActivationKey(activationKey);
+      setRole(data?.role);
+      setServiceLevel(data?.serviceLevel);
+      setUsage(data?.usage);
+      setRepositories(data?.additionalRepositories);
+    };
+    fetchKeyInformation();
+  }, []);
+
+  return (
+    <>
+      <TextContent>
+        <TextList component={TextListVariants.dl}>
+          <TextListItem component={TextListItemVariants.dt}>Name:</TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {activationKey}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>Role:</TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {role || 'Not defined'}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>SLA:</TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {serviceLevel || 'Not defined'}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>
+            Usage:
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dd}>
+            {usage || 'Not defined'}
+          </TextListItem>
+          <TextListItem component={TextListItemVariants.dt}>
+            Additional <br /> repositories:
+            <Popover
+              bodyContent={
+                <TextContent>
+                  <Text>
+                    The core repositories for your operating system version are
+                    always enabled and do not need to be explicitly added to the
+                    activation key.
+                  </Text>
+                </TextContent>
+              }
+            >
+              <Button
+                variant="plain"
+                aria-label="About additional repositories"
+                className="pf-u-pl-sm pf-u-pt-0 pf-u-pb-0"
+                isSmall
+              >
+                <HelpIcon />
+              </Button>
+            </Popover>
+          </TextListItem>
+          <TextListItem
+            component={TextListItemVariants.dd}
+            className="pf-u-display-flex pf-u-align-items-flex-end"
+          >
+            {additionalRepositories?.length === 0 ? (
```

Comment:
If `additionalRepositories` is `undefined` for whatever reason, then we will render the `<Popover>` component... which I don't think we want. What if this was something like: `additionalRepositories?.length > 0 ? (<Popover ... />) : ('None')`? That way, we only render the popover with the additional repositories table if we are 100% sure there are additional repositories.

Reply:
Fixed :+1: